src/plugins/data/public/ui/dataset_selector/index.tsx

```diff
@@ -57,7 +54,7 @@ const ConnectedDatasetSelector = ({
     <DatasetSelector
       {...datasetSelectorProps}
       selectedDataset={selectedDataset}
-      setSelectedDataset={handleDatasetChange}
+      onSelect={onSelect}
```

Comment:
Nit: 
```suggestion
      onChance={handleChange}
```

This is a more common pattern

Reply:
can make it a fast follow?